src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -48,6 +49,8 @@ type DownsampleAndWriteIter interface {
 	Current() (models.Tags, ts.Datapoints, xtime.Unit, []byte)
 	Reset() error
 	Error() error
+	SetCurrentMetadata(ts.Metadata)
+	GetCurrentMetadata() ts.Metadata
```

Comment:
As per Effective Go we should keep SetCurrentMetadata but remove the "Get" prefix and just have the getter as "CurrentMetadata()":
> There's nothing wrong with providing getters and setters yourself, and it's often appropriate to do so, but it's neither idiomatic nor necessary to put Get into the getter's name. If you have a field called owner (lower case, unexported), the getter method should be called Owner (upper case, exported), not GetOwner. The use of upper-case names for export provides the hook to discriminate the field from the method. A setter function, if needed, will likely be called SetOwner. Both names read well in practice

https://golang.org/doc/effective_go.html#Getters